### PR TITLE
Release v0.3.3

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: 0.3.3

Autonomy:

- Introduces support for specifying tag versions for runtime image builds.
- Adds the benchmark tool definition in the newly scaffolded FSM skill.
- Adds improvements on the FSM scaffolding
- Adds support for ACN and hardhat node in deployment setup using build flags.
- Fixes regex in analyse to avoid capturing empty strings.

Tests:

- Adds tests to make sure that scaffolded FSM modules can be tested using the CLI command `autonomy test`.
- Adds test coverage for the app spec module.
- Cleans up various test class usage across `tests/test_autonomy/test_cli`.

Packages:

- Ports following packages to their respective github repositories
  - APY skills, agent, and service
  - Liquidity rebalancing skills.
  - Price oracle skills, agent, and service
- Fixes the URL check on tendermint handler

Docs:

- Improves to quickstart section. 
- Improves and on-chain protocol guides.
- Simplifies and cleans up the demo sections.
- Changes the images to follow the new colour scheme.
- Removes redundant information and redirects to corresponding documentation sections.
- Replaces the usage of oracle service in CLI documentation with the hello world service. 

Chores:

- Fixes make target for image release to avoid issues on different shells.
- Updates some scripts that perform multiple network requests to use parallelization.
- Restructures `Running on other networks` section.

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] I've updated the dependencies versions to the latest, wherever is possible.
- [ ] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [ ] I built the documentation and updated it with the latest changes
- [ ] I've added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

### CODE REVIEW

File: .spelling [hunk 1]
```diff
@@ -206,6 +206,10 @@ faucet
 OpenSea
 metadada
 GörliWei
+scaffolded
+spec
+github
```
Comment (DavidMinarsch) [inline .spelling:211]: It should be `GitHub`, hence fix and remove here.

It also should be `specification`

File: HISTORY.md [hunk 2]
```diff
@@ -1,6 +1,46 @@
 # Release History - `open-autonomy`
 
 
+# 0.3.3 (2022-10-21)
+
+Autonomy:
+
+- Introduces support for specifying tag versions for runtime image builds.
+- Adds the benchmark tool definition in the newly scaffolded FSM skill.
+- Adds improvements on the FSM scaffolding
```
Comment (DavidMinarsch) [inline HISTORY.md:10]:
```suggestion
- Adds improvements on the FSM scaffolding.
```

File: HISTORY.md [hunk 3]
```diff
@@ -1,6 +1,46 @@
...
+Tests:
+
+- Adds tests to make sure that scaffolded FSM modules can be tested using the CLI command `autonomy test`.
+- Adds test coverage for the app spec module.
+- Cleans up various test class usage across `tests/test_autonomy/test_cli`.
+
+Packages:
+
+- Ports following packages to their respective github repositories
+  - APY skills, agent, and service
```
Comment (DavidMinarsch) [inline HISTORY.md:23]:
```suggestion
  - APY skills, agent, and service;
```

File: HISTORY.md [hunk 4]
```diff
@@ -1,6 +1,46 @@
...
+  - Liquidity rebalancing skills.
+  - Price oracle skills, agent, and service
+- Fixes the URL check on tendermint handler
```
Comment (DavidMinarsch) [inline HISTORY.md:26]:
```suggestion
- Fixes the URL check on the tendermint handler in `abstract_round_abci`.
```

File: HISTORY.md [hunk 5]
```diff
@@ -1,6 +1,46 @@
...
+
+Docs:
+
+- Improves to quickstart section.
```
Comment (DavidMinarsch) [inline HISTORY.md:30]:
```suggestion
- Improves the quickstart section. 
```

File: HISTORY.md [hunk 6]
```diff
@@ -1,6 +1,46 @@
...
+- Ports following packages to their respective github repositories
+  - APY skills, agent, and service
+  - Liquidity rebalancing skills.
+  - Price oracle skills, agent, and service
+- Fixes the URL check on tendermint handler
+
+Docs:
+
+- Improves to quickstart section. 
+- Improves and on-chain protocol guides.
```
Comment (DavidMinarsch) [inline HISTORY.md:31]:
```suggestion
- Improves the on-chain protocol guides.
```

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
LGTM, except many minor typos

**Review by Adamantios — CHANGES_REQUESTED** [review]
Update here too:
https://github.com/valory-xyz/open-autonomy/blob/e18f685981c9f6a7fd785d1f355c0cac3eb86982/tests/test_base.py#L65

**Review by DavidMinarsch — APPROVED** [review]

**Review by Adamantios — APPROVED** [review]

**Review by Karrenbelt — APPROVED** [review]
LGTM

**Pull request merged**